wannierberri/__utility.py

```diff
@@ -284,15 +287,51 @@ def __call__(self,AAA_R,hermitian=False,antihermitian=False,reshapeKline=True):
                         for k[2] in range(self.NKFFT[2]) ] for k[1] in range(self.NKFFT[1]) ] for k[0] in range(self.NKFFT[0])  ] )
-            t=time()-t0
-#            print ("slow FT finished in {} sec for AAA_R {} and {} k-grid . {} per element".format(t,AAA_R.shape,self.NKFFT,t/np.prod(self.NKFFT )/np.prod(AAA_R.shape)))
+                #print('AAA_K',np.shape(AAA_K))
+                t=time()-t0
+            else:
+                print('convention 1 slow FT')
+                t0=time()
+                #print(np.round(self.wannier_centres,decimals=2))
+                w_centres = np.array([[j-i for j in self.wannier_centres] for i in self.wannier_centres])
+                exponent=[np.exp(2j*np.pi/self.NKFFT[i])**np.arange(self.NKFFT[i]) for i in range(3)]
+                def exp_par(ii,jj,i):#k1 k2 k3 partial of exponent_wc
+                    return np.prod(np.exp(2j*np.pi*w_centres[ii,jj,i]/self.NKFFT[i])**np.arange(self.NKFFT[i]))
+                exponent_wc=np.array([[exp_par(ii,jj,0)*exp_par(ii,jj,1)*exp_par(ii,jj,2)
+                            for jj in range(self.num_wann)] for ii in range(self.num_wann)])
+                k=np.zeros(3,dtype=int)
+                dig_w_centres = 0.0
+                if len(np.shape(AAA_R)) == 4: ###TODO only for AA_R now. Change it suit every matrix later. 
+                    # A dig matrix delta_ij*tau_i
+                   # print(AAA_R[172,:,:,0].real)
+                    #print(np.shape(AAA_R))
+                    dig_w_centres = np.zeros((self.num_wann,self.num_wann,3))
+                    for i in range(self.num_wann):
+                        dig_w_centres[i,i,:] = self.wannier_centres[i,:]
+                    exponent_wc=exponent_wc[:,:,None]
+                    #print(np.round(AAA_R,decimals=2)[:,:,120,0].real)
+                    #print(np.round(dig_w_centres,decimals=2)[:,:,0])
+                if len(np.shape(AAA_R)) == 5:
+                    exponent_wc=exponent_wc[:,:,None,None]
+                #print('AAA_R',np.shape(AAA_R))
+                #print(np.shape(dig_w_centres))
+                AAA_K=np.array([[[
+                    sum( np.prod([exponent[i][(k[i]*R[i])%self.NKFFT[i]] for i in range(3)]) * A   for R,A in zip( self.iRvec, AAA_R) )* exponent_wc - dig_w_centres 
+                    #sum( np.prod([exponent[i][(k[i]*R[i])%self.NKFFT[i]] for i in range(3)]) 
+                    #    * [np.prod(exponent[i][k[i]])**w_centres[:,:,i] for i in range(3)] * A    for R,A in zip( self.iRvec, AAA_R) )
+                    for k[2] in range(self.NKFFT[2]) ] for k[1] in range(self.NKFFT[1]) ] for k[0] in range(self.NKFFT[0])  ] )
+                #print('AAA_K',np.shape(AAA_K))
+                t=time()-t0
+            #print ("slow FT finished in {} sec for AAA_R {} and {} k-grid . {} per element".format(t,AAA_R.shape,self.NKFFT,t/np.prod(self.NKFFT )/np.prod(AAA_R.shape)))
```

Comment:
Because the modifications are independent of the way of doing Fourier transform (as follows from my handwritten note) all changes should be made below, just before `if hermitian:`